### PR TITLE
added add function

### PR DESCRIPTION
**Pull request by DavideTisi**
A first draft of a sum function, so far it has no test.
it sums a TensorMap + a constant value/array or TensorMap + TensorMap. 
In the first case, the metadata of the result is the one of th; input TensorMap, in the second the two input TensorMaps must have the same metadata that will be the same as the input.

### CODE REVIEW

**Comment by DavideTisi** [comment]
the sum with array should be not possible

File: python/src/equistore/operations/multiply.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+from ..block import TensorBlock
+from ..tensor import TensorMap
+from ._utils import _check_blocks, _check_same_gradients_components, _check_same_keys
+
+
+def add(A: TensorMap, B) -> TensorMap:
```
Comment (PicoCentauri) [inline python/src/equistore/operations/multiply.py:6]:
```suggestion
def multiply(A: TensorMap, B) -> TensorMap:
```

**Review by PicoCentauri — COMMENTED** [review]
Just one minor comment and no real reveiw so far. Most importantly, I think we should also at least add the [magic functions](https://docs.python.org/3/reference/datamodel.html#emulating-numeric-types) 
`__add__()` and `__mul__()`   for the `TensorMap` and `TensorBlock` class.

But, I can also do this in a upcoming PR...

**Comment by DavideTisi** [comment]
I added the sum and multiplication for scalar and among `TensorMaps` with the same exact metadata. This last bit is not general but I would start merging this PR given the need for addition/multiplication with scalar. An to complete the general operations later.

**Comment by PicoCentauri** [comment]
@DavideTisi I did some changes to the docstring and to the code to make a bit clearer. If you agree you can rebase/squash and merge.

**Review by PicoCentauri — APPROVED** [review]

File: docs/src/reference/python/operations/add.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+join
+====
```
Comment (Luthaf) [inline docs/src/reference/python/operations/add.rst:2]:
```suggestion
add
===
```

File: docs/src/reference/python/operations/multiply.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+join
+====
```
Comment (Luthaf) [inline docs/src/reference/python/operations/multiply.rst:2]:
```suggestion
multiply
========
```

File: python/src/equistore/operations/add.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            raise TypeError("B must be a TensorMap or a float.") from e
+        for blockA in A.blocks():
+            blocks.append(_add_block_constant(block=blockA, constant=B))
+
+    return TensorMap(A.keys, blocks)
+
+
+def _add_block_constant(block: TensorBlock, constant: float) -> TensorBlock:
+
+    # values = block1.values @ block2.values.T
```
Comment (Luthaf) [inline python/src/equistore/operations/add.py:50]:
```suggestion
```

File: python/src/equistore/operations/multiply.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    """Return a new :class:`TensorMap` with the values being the element-wise
+    multiplication of ``A`` and ``B``.
+
+    If ``B`` is a :py:class:`TensorMap` it has to have the same metadata as ``A``.
+
+    If gradients are present in ``A`` a multiplication is only performed if ``B`` is
+    a :py:class:`TensorMap` as well. The multiplication is performed with the 
+    rule of the derivatives::
+
+      TODO
```
Comment (Luthaf) [inline python/src/equistore/operations/multiply.py:21]: ?

**Pull request merged**